### ISSUE TITLE
DesperateDevs.CodeGeneration.Plugins.ProjectPath only supports one project.

### ISSUE THREAD

**Issue by bfsrjose** [sschmid/Entitas#682]
Hi,

I have a suggestion:

DesperateDevs.CodeGeneration.Plugins.ProjectPath only supports one project, supporting a list would be helpful.

Cheers

**Comment by sschmid** [comment]
Basically, if I add optional meta data to the CodeGenFile, e.g. project path, then the post processor can update the corresponding project

**Comment by bfsrjose** [comment]
Awesome :-)

**Comment by sschmid** [comment]
Hi,
in order to make it work as I imagined, each data provider would have to specify the project path, so the post processor can map the `CodeGenFile` to the correct project. This would be a big change for an unusual case.

If you already found a way how to make it work for your setup, I'd prefer not to change the existing implementation.

Or do you have another idea how this could be solved?

**Comment by bfsrjose** [comment]
Well, you can maintain all paths relative to the properties directory (or make them absolute) and then adjust them shortly before writing to the project file ~ assuming at that point you know where the project file you are writing to is.

**Comment by c0ffeeartc** [comment]
Multiple projects could be handled with multiple passes. Keep separate sets of setting files per project, and buttons `Generate` and `Generate All Passes`

**Comment by sschmid** [comment]
Fyi, I'm working on a new code generator using dotnet's `IIncrementalGenerator`.

tldr:
this will allow propper namespace support + multiple projects, e.g. using Unity's asmdef

Will close this issue, but you can follow the progress here:
#1005
